### PR TITLE
Guide to get NFTs Owned by an Account

### PR DESCRIPTION
**Pull request by Ginowine**
# Description of change

Created a guide to demonstrate using the `getL2NFTs` function from the `ISCAccounts.sol` interface. It explains the function's purpose, operation within the `ISCAccounts` interface, and practical applications like portfolio management and marketplace listings.

## Links to any relevant issues

fixes issue #1630 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Ginowine** [comment]
@lucas-tortora, this is a PR awaiting your awesomeness review, comments, suggestions, and any improvement you think is needed.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   Ensure you have the ISC Magic library installed in your project:
+   ```bash
+   npm install @iota/iscmagic
+   ```
+
+2. **Import the ISC Library:**
+
+   In your Solidity file, import the ISC Library:
+   ```solidity
+   pragma solidity >=0.8.5;
```
Comment (salaheldinsoliman) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:44]: Thanks for limiting the contract version!

**Review by salaheldinsoliman — APPROVED** [review]

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 1 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:44]: Thanks, I'll implement this in my next commit, after receiving any suggestions from others. I appreciate your time.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+---
+description: How to Get the L2 NFTs owned by an account.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:2]:
```suggestion
description: How to get the L2 NFTs owned by an account.
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  - EVM
+  - how-to
+  - native tokens
+  - L2 NFTs
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFTs Owned by an Account
+
+This guide will show you how to get the L2 NFTs owned by an account using the `getL2NFTs` function from the `ISCAccounts` magic contract.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:14]:
```suggestion
This guide will show you how to get the L2 NFTs owned by an account using the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md#getl2nfts) function from the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md) [magic contract](../../../reference/magic-contract/introduction.md).
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+<GetNftMetadata />
+
+## Understanding the `getL2NFTs` Function
+
+The `getL2NFTs` function is a view function provided by the `ISCAccounts` interface. It takes an `ISCAgentID` as an argument and returns an array of `NFTID` objects. This function allows users to query and retrieve the list of L2 NFTs owned by a specified agent. The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. The returned array contains the identifiers of the NFTs, which can then be used to fetch more details about each NFT if needed.
+
+## Why Would a User Need to retrieve owned NFTs?
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:22]:
```suggestion
## Why Would a User Need to Retrieve Owned NFTs?```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+Retrieving the NFTs owned by a user can serve several purposes:
+
+1. **Portfolio Management:** Users can manage their digital assets effectively by viewing all their NFTs in one place.
+2. **Marketplace Listing:** Users can list their NFTs on various marketplaces by verifying their ownership.
+3. **Collection Display:** Users can showcase their NFT collections on social media, personal websites, or within specific applications.
+4. **Trading and Transfers:** Knowing the NFTs owned helps users decide which assets to trade or transfer to other users.
+5. **Application Integration:** Applications can provide customized experiences based on the NFTs a user owns, such as granting access to special content or features.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:30]:
```suggestion
### Portfolio Management

Users can manage their digital assets effectively by viewing all their NFTs in one place.

### Marketplace Listing

Users can list their NFTs on various marketplaces by verifying their ownership.

### Collection Display

Users can showcase their NFT collections on social media, personal websites, or within specific applications.

### Trading and Transfers

Knowing the NFTs owned helps users decide which assets to trade or transfer to other users.

### Application Integration

Applications can provide customized experiences based on the NFTs a user owns, such as granting access to special content or features.

```

Alternatively, make this an unordered list, as it's not instructions or a ranking.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Prerequisites
+1. **Install the ISC Magic Library:**
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:33]:
```suggestion

### Install the ISC Magic Library

```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+   Ensure you have the ISC Magic library installed in your project:
+   ```bash
+   npm install @iota/iscmagic
+   ```
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:38]:
````suggestion
Ensure you have the ISC Magic library installed in your project:

```bash
npm install @iota/iscmagic
```
````

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+2. **Import the ISC Library:**
+
+   In your Solidity file, import the ISC Library:
+   ```solidity
+   pragma solidity >=0.8.5;
+
+   import "@iota/iscmagic/ISC.sol";
+   ```
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:47]:
````suggestion
### 2. Import the ISC Library

In your Solidity file, import the ISC Library:
```solidity
pragma solidity >=0.8.24;

import "@iota/iscmagic/ISC.sol";
```

````

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+function getOwnedNFTs(ISCAgentID memory agentID) public view returns (NFTID[] memory) {
+    // Call the getL2NFTs function from the ISCAccounts contract
+    NFTID[] memory ownedNFTs = ISC.accounts.getL2NFTs(agenID);
+    return ownedNFTs;
+    }
+}
+```
+## Conclusion
+
+Using the `getL2NFTs` function from the `ISCAccounts` interface provides an efficient way to retrieve and manage the Layer 2 NFTs owned by an account. By integrating this functionality into your smart contracts, you can enhance user experience by enabling features such as portfolio management, marketplace listings, collection displays, and more. With this guide, you now have the foundational knowledge to implement and utilize the `getL2NFTs` function in your own projects, allowing for seamless interaction with `ISC Magic` and its associated NFTs.
```
Comment (lucas-tortora) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:79]:
```suggestion
Using the `getL2NFTs` function from the `ISCAccounts` interface provides an efficient way to retrieve and manage the Layer 2 NFTs owned by an account. By integrating this functionality into your smart contracts, you can enhance user experience by enabling features such as [portfolio management](#portfolio-management), [marketplace listings](#marketplace-listing), [collection displays](#collection-display), and more. With this guide, you have the foundational knowledge to implement and utilize the `getL2NFTs` function in your projects, allowing for seamless interaction with `ISC Magic` and its associated NFTs.
```

**Review by lucas-tortora — COMMENTED** [review]
Just some comments regarding format, great article :)

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 3 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: Thanks for the suggestion, done.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 5 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: Ok, great. Please look at how I implemented this based on your alternative suggestion.

**Comment by Ginowine** [comment]
@lucas-tortora, I have implemented your suggestions. Thanks for sharing them.

**Review by lucas-tortora — APPROVED** [review]

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+# Get NFTs Owned by an Account
+
+This guide will show you how to get the L2 NFTs owned by an account using the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md#getl2nfts) function from the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md) [magic contract](../../../reference/magic-contract/introduction.md).
+
+<GetNftMetadata />
+
+## Understanding the `getL2NFTs` Function
+
+The `getL2NFTs` function is a view function provided by the `ISCAccounts` interface. It takes an `ISCAgentID` as an argument and returns an array of `NFTID` objects. This function allows users to query and retrieve the list of L2 NFTs owned by a specified agent. The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. The returned array contains the identifiers of the NFTs, which can then be used to fetch more details about each NFT if needed.
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:20]: Maybe we should add an admonition here on what an AgentID exactly is and link to more content.

We should probably also add How-Tos on how to do the most common type conversions. Like address to agentid or NFTID to token and so on

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+  - Users can showcase their NFT collections on social media, personal websites, or within specific applications.
+
+- ### Trading and Transfers
+
+  - Knowing the NFTs owned helps users decide which assets to trade or transfer to other users.
+
+- ### Application Integration
+
+  - Applications can provide customized experiences based on the NFTs a user owns, such as granting access to special content or features.
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:44]:
```suggestion
```

I would keep the how-tos on point. Not too much stuff around it. We can create user stories in tutorials imo

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+### 2. Import the ISC Library
+
+   In your Solidity file, import the ISC Library:
+
+   ```solidity
+   pragma solidity >=0.8.24;
+
+   import "@iota/iscmagic/ISC.sol";
+   ```
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:64]: Let's not duplicate content. We explain this I the magic contract introduction
```suggestion
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+contract MyNFTContract {
+
+function getOwnedNFTs(ISCAgentID memory agentID) public view returns (NFTID[] memory) {
+    // Call the getL2NFTs function from the ISCAccounts contract
+    NFTID[] memory ownedNFTs = ISC.accounts.getL2NFTs(agenID);
+    return ownedNFTs;
+    }
+}
+```
+## Conclusion
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx:95]: I like consistency, so I would remove it as we don't have it in other how-tos either. But no strong opinion from my side. Question is if someone will read it after the full how-to

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
Great to see this documented. Already helped one of our partners. Great job.

Just some smaller changes.
Don't forget to apply my changes on both pages

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+  - EVM
+  - how-to
+  - native tokens
+  - L2 NFTs
+---
+import GetNftMetadata from '../../../_partials/how-tos/token/_get-nft-metadata.md';
+
+# Get NFTs Owned by an Account
+
+This guide will show you how to get the L2 NFTs owned by an account using the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md#getl2nfts) function from the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md) [magic contract](../../../reference/magic-contract/introduction.md).
```
[the remark this answers is not in the document]
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: Thanks, updated.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 11 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: Ok, removed.

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 10 above)
Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: @Dr-Electron, I added an admonition, but I don't know if the info about ISCAgentID is enough. I think the URL I referenced in the admonition might not be the best.

Comment (Ginowine) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: Or maybe you want me to remove this enter section and just leave the admonition?

File: docs/build/isc/v1.1/docs/_admonitions/_AgentID.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+:::Info ISC Agent ID
+The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. You can get the [`AgentID`](../explanations/how-accounts-work.md) from the sender by calling `ISC.sandbox.getSenderAccount()`.
+:::
```
Comment (Dr-Electron) [inline docs/build/isc/v1.1/docs/_admonitions/_AgentID.md:3]:
```suggestion
:::info Agent ID

In ISC [Agent IDs](../explanations/how-accounts-work.md) are used to encode L1 and L2 addresses in a unique way as same L2 addresses can exist on different ISC chains.

:::
```

Comment (lucas-tortora) [reply docs/build/isc/v1.1/docs/_admonitions/_AgentID.md:3]:
```suggestion
:::Info ISC Agent ID

The `ISCAgentID` is a unique identifier for any type of account. In ISC, [accounts may represent addresses from L1, L2, or even smart contracts](../explanations/how-accounts-work.md). You can retrieve the sender's `Agent Id` by calling `ISC.sandbox.getSenderAccount()`.

:::
```

Comment (lucas-tortora) [reply docs/build/isc/v1.1/docs/_admonitions/_AgentID.md:3]: The original sounds good tbh

Comment (Dr-Electron) [reply docs/build/isc/v1.1/docs/_admonitions/_AgentID.md:3]: But I still have a change:
```suggestion
:::info ISC Agent ID

The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. You can get the [`AgentID`](../explanations/how-accounts-work.md) from the sender by calling `ISC.sandbox.getSenderAccount()`.

:::
```

File: docs/build/isc/v1.3/docs/_admonitions/_AgentID.md [hunk 16]
```diff
@@ -0,0 +1,3 @@
+:::info ISC Agent ID
+The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. You can get the [`AgentID`](../explanations/how-accounts-work.md) from the sender by calling `ISC.sandbox.getSenderAccount()`.
+:::
```
Comment (Dr-Electron) [inline docs/build/isc/v1.3/docs/_admonitions/_AgentID.md:3]:
```suggestion
:::info ISC Agent ID

The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. You can get the [`AgentID`](../explanations/how-accounts-work.md) from the sender by calling `ISC.sandbox.getSenderAccount()`.

:::
```

File: docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx (hunk 10 above)
Comment (Dr-Electron) [reply docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx]: It is good enough for now. Lets see if we want to change it later or extend the linked article in another iteration

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**